### PR TITLE
Update bindings: v2.0.16

### PR DESCRIPTION
**Pull request by chrisimcevoy**
- Update bindings to v2.0.16 (API v1.566.2)
- Introduces a swagger-codegen-cli image to replace the .jar files previously stored in LFS.
- Takes the latest 2.4.x version of swagger-codegen-cli which includes a fix for the generated python client which previously added a basic auth Authorization header with every request, even when a username and password was not configured.
- Adds `.idea` to `.gitignore` to prevent IDE-specific files from being committed.

For more details on the forementioned bug and fix:
- https://github.com/swagger-api/swagger-codegen/issues/9430
- https://github.com/swagger-api/swagger-codegen/pull/12212

This PR relates to:
- #58

### CODE REVIEW

File: scripts/common.sh [hunk 1]
```diff
@@ -3,7 +3,8 @@ language=${1:-""}
 api_url=${2:-${api_url:-"https://api.cloudsmith.io/"}}
 api_version=$(curl -s "${api_url}status/check/basic/" | jq -r '.version')
 openapi_url="${api_url}?format=openapi"
-package_version="2.0.15"
+package_version="2.0.16"
+swagger_codegen_cli_image="swaggerapi/swagger-codegen-cli:2.4.43"
```
Comment (chrisimcevoy) [inline scripts/common.sh:7]: A new variable for the swagger-codegen-cli image we wish to use. This is used in the main build.sh script, and in the build.sh scripts nested inside the $lang directories.

**Review by lskillen — APPROVED** [review]
Glad to see the auth bug squashed. 😁

**Review by ralph-mcteggart — APPROVED** [review]
Amazing 💪

**Comment by nickbroon** [comment]
As this now pulling a swagger-codegen-cli docker container to use while building the bindings, can the [/bin](https://github.com/cloudsmith-io/cloudsmith-api/tree/master/bin) directory can now be removed from the repository? (which should reduce its cloned size).

**Pull request merged**